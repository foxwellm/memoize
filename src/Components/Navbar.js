import React, { Component } from 'react';

export default class Navbar extends Component {
  constructor() {
    super()
    this.state = {
      currentActivePage: 'string'
    }
  }

  changePage = (event) => {
    this.setActive(event)
    this.props.setSlideshowPage(event)
  }

  setActive = (event) => {
    let currentActivePage = event.target.dataset.page
    this.setState({
      currentActivePage
    })
  }

  btnClasses(btn) {
    const cssClasses = [
      "nav-btn",
      this.state.currentActivePage === btn ? "active-btn" : null
    ].join(' ')
    return cssClasses
  }


  render() {
    const { favorites } = this.props
    let currentFavoriteCount = 0;
    Object.keys(favorites).forEach(key => {
      currentFavoriteCount += favorites[key].length
    })

    return (
      <div className="nav-container">
        <div className={this.btnClasses('homepage')} onClick={this.changePage} data-page={'homepage'} >MEMOIZE</div>
        <div className={this.btnClasses('string')} onClick={this.changePage} data-page={'string'}>String</div>
        <div className={this.btnClasses('array')} onClick={this.changePage} data-page={'array'}>Array</div>
        <div className={this.btnClasses('favorites')} onClick={this.changePage} data-page={'favorites'}>Favorites ({currentFavoriteCount})</div>
      </div>
    )
  }
}