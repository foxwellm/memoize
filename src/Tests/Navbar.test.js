import React from 'react';
import Navbar from '../Components/Navbar';
import { shallow } from 'enzyme';

describe('Navbar', () => {
  let wrapper;
  const setSlideshowPageMock = jest.fn();
  const currentPage = 'homepage';
  const favorites = { string: ["concat", "charCodeAt"], array: ["every", "forEach"] }
  const event = {
    target: {
      dataset: {
        page: 'string'
      },
    }
  }

  beforeEach(() => {

    wrapper = shallow(
      <Navbar
        setSlideshowPage={setSlideshowPageMock}
        currentPage={currentPage}
        favorites={favorites}
      />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should set active state of button and change to clicked page', () => {
    expect(wrapper.state('currentActivePage')).toEqual('homepage');
    wrapper.instance().changePage(event);
    expect(wrapper.state('currentActivePage')).toEqual('string');
  })

})