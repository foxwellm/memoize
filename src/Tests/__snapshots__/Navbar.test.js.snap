// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar
    currentPage="homepage"
    favorites={
      Object {
        "array": Array [
          "every",
          "forEach",
        ],
        "string": Array [
          "concat",
          "charCodeAt",
        ],
      }
    }
    setSlideshowPage={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="nav-btn active-btn"
          data-page="homepage"
          onClick={[Function]}
        >
          MEMOIZE
        </div>,
        <div
          className="nav-btn "
          data-page="string"
          onClick={[Function]}
        >
          String
        </div>,
        <div
          className="nav-btn "
          data-page="array"
          onClick={[Function]}
        >
          Array
        </div>,
        <div
          className="nav-btn "
          data-page="favorites"
          onClick={[Function]}
        >
          Favorites (
          4
          )
        </div>,
      ],
      "className": "nav-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "MEMOIZE",
          "className": "nav-btn active-btn",
          "data-page": "homepage",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "MEMOIZE",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "String",
          "className": "nav-btn ",
          "data-page": "string",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "String",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Array",
          "className": "nav-btn ",
          "data-page": "array",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Array",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Favorites (",
            4,
            ")",
          ],
          "className": "nav-btn ",
          "data-page": "favorites",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Favorites (",
          4,
          ")",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="nav-btn active-btn"
            data-page="homepage"
            onClick={[Function]}
          >
            MEMOIZE
          </div>,
          <div
            className="nav-btn "
            data-page="string"
            onClick={[Function]}
          >
            String
          </div>,
          <div
            className="nav-btn "
            data-page="array"
            onClick={[Function]}
          >
            Array
          </div>,
          <div
            className="nav-btn "
            data-page="favorites"
            onClick={[Function]}
          >
            Favorites (
            4
            )
          </div>,
        ],
        "className": "nav-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "MEMOIZE",
            "className": "nav-btn active-btn",
            "data-page": "homepage",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "MEMOIZE",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "String",
            "className": "nav-btn ",
            "data-page": "string",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "String",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Array",
            "className": "nav-btn ",
            "data-page": "array",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Array",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Favorites (",
              4,
              ")",
            ],
            "className": "nav-btn ",
            "data-page": "favorites",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Favorites (",
            4,
            ")",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
